peercrawler.py

```diff
@@ -27,19 +24,22 @@ def get_telemetry(ctx, s):
 
 
 class peer_manager:
-    def __init__(self, ctx, peers=[], verbosity=0):
+    def __init__(self, ctx, peers=[], verbosity=0, inactivity_threshold_seconds=0):
         self.ctx = ctx
-        self.mutex = threading.Lock()
-        self.peers = set()
         self.verbosity = verbosity
+        self.inactivity_threshold_seconds = inactivity_threshold_seconds
+
+        self.mutex = threading.Lock()
+        # a dictionary storing all the discovered peers and the last time it was seen
+        self.peers: dict[Peer, int] = {}
```

Comment:
I wonder what is better, to do this way, or to keep the last time a peer was seen as an attribute inside the peer?